lib/Reader/readerir.cpp

```diff
@@ -723,6 +690,16 @@ void GenIR::insertIRForUnmanagedCallFrame() {
 //
 //===----------------------------------------------------------------------===//
 
+AllocaInst *GenIR::createAlloca(Type *T, Value *ArraySize, const Twine &Name) {
+  AllocaInst *AllocaInst = LLVMBuilder->CreateAlloca(T, ArraySize, Name);
+
+  if (JitContext->Options->DoInsertStatepoints && GcInfo::isGcAggregate(T)) {
+    GcFuncInfo->recordGcAggregate(AllocaInst);
+  }
```

Comment:
I left the DoInsertStatepoints check here because even if the information is recorded on GcFuncInfo, it will not be reported for conservative GC (GcInfoEmitter::needsPointerReporting() will be false).

Conservative GC should not need this reported, because it must treat all those values as pointers already.

But I can remove the check, if you think its better to simplify the reader.